### ISSUE TITLE
Possible issue with UCRT runtime: directory creation permissions in .emacs.d

### ISSUE THREAD

**Issue by the42** [kiennq/emacs-build#3]
I experience the following issue since this Emacs build switched to the UCRT so since two months: directories in the .emacs.d configuration directory are created without write permissions. This results in directory deletion error when eg. elpa tries to delete outdated packages or cache in the url directory. In the past, using the "old" migw c-runtime, this was not an issue.

The error in *Message* is: 

Error when expiring the cache: (permission-denied Removing directory Permission denied c:/Users/HoechtlJ/OneDrive - WKO Online/Dokumente/.emacs.d/url/cache/HoechtlJ/https/org/gnu/www)

If I manuelly chmod -R +w <directory> the  problem disappears.

The home directory is located on a onedrive folder and synched into the cloud. So this may coincide with a change how onedrive syncs files/directoies.

**Comment by kiennq** [comment]
Can you try `--init-directory` parameter to launch Emacs to see if it's because OneDrive issue or Emacs issue?

**Comment by the42** [comment]
I was not aware of this setting. I tried it out. I first copyied my .emacs.d to a location not monitored by OneDrive and set --init-directory afterwards accordingly.


I browsed with eww the Gnu homepage and indeed, folders created on a non-Onedrive monitored directory are created with +w set.

I then tried opening emacs with .emacs.d on the OneDrive monitored $HOME, also used eww to browse the Gnu homepage and the cache directories are create WITHOUT the chmod -w flag.

So this confirms that directory creation on a OneDrive synched .emacs.d results in incorrect directory permissions.
